File: src/wp7/Meet4Xmas/org/meet4xmas/wire/TravelPlan.cs
using System.Text;

namespace org.meet4xmas.wire
{
    public class TravelPlan
    {
        public static class TravelType
        {
            public sealed static const int Car = 0;
            public sealed static const int Walk = 1;
            public sealed static const int PublicTransport = 2;

            public static string ToString(int travelType)
            {
                switch (travelType)
                {
                    case Car: return "Car";
                    case Walk: return "Walk";
                    case PublicTransport: return "PublicTransport";
                    default: return "Invalid";
                }
            }
        }

        public Location[] path;

        public override string ToString() {
            StringBuilder sb = new StringBuilder("<TravelPlan ");
            sb.Append("@path: ").Append(path);
            sb.Append(">");
            return sb.ToString();
        }
    }
}

File: src/android/src/org/meet4xmas/wire/TravelPlan.java
package org.meet4xmas.wire;

import org.meet4xmas.util.StringUtil;

public class TravelPlan implements java.io.Serializable {
  public static class TravelType {
    public final static int Car = 0;
    public final static int Walk = 1;
    public final static int PublicTransport = 2;

    public static String toString(int travelType) {
      switch(travelType) {
        case Car: return "Car";
        case Walk: return "Walk";
        case PublicTransport: return "PublicTransport";
        default: return "Invalid";
      }
    }
  }

  public Location[] path;

  public String toString() {
    StringBuilder sb = new StringBuilder("TravelPlan(");
    sb.append("@path: ").append(StringUtil.ArrayOrNullToString(path));
    sb.append(")");
    return sb.toString();
  }
}
